### PR TITLE
Update Order for v1/v3

### PR DESCRIPTION
**Pull request by Joseph-Edwards**
This PR adds some comparison functions, and documentation for the Order class.

### CODE REVIEW

File: src/order.cpp [hunk 1]
```diff
@@ -45,14 +35,172 @@ namespace py = pybind11;
...
+:param x: the first object for comparison. 
+:type x: str | List[int]
+
+:param y: the second object for comparison.
+:type y: str | List[int]
+
+:returns: The boolean value ``True`` if *x* is less than *y* with respect to the recursive path ordering, and ``False`` otherwise.
+:rtype: bool
+
+:exceptions: This function is and is guaranteed never to throw.
```
Comment (james-d-mitchell) [inline src/order.cpp:189]: This line is sus

File: src/order.cpp [hunk 2]
```diff
@@ -45,14 +35,172 @@ namespace py = pybind11;
...
+:type x: str | List[int]
+
+:param y: the second object for comparison.
+:type y: str | List[int]
+
+:returns: The boolean value ``True`` if *x`* is short-lex less than *y*, and ``False`` otherwise.
+:rtype: bool
+
+:exceptions: This function guarantees not to throw a ``LibsemigroupsError``.
+
```
Comment (james-d-mitchell) [inline src/order.cpp:139]: I think we weren't including these lines in the python doc

File: src/order.cpp [hunk 3]
```diff
@@ -45,14 +35,172 @@ namespace py = pybind11;
...
+:param x: the first object for comparison. 
+:type x: str | List[int]
+
+:param y: the second object for comparison.
+:type y: str | List[int]
+
+:returns: The boolean value ``True`` if *x* is lexicographically less than *y*, and ``False`` otherwise.
+:rtype: bool
+
+:exceptions: This function guarantees not to throw a ``LibsemigroupsError``.
```
Comment (james-d-mitchell) [inline src/order.cpp:100]: Ditto

**Review by james-d-mitchell — CHANGES_REQUESTED** [review]
No couple of quibbles, then merge

File: src/order.cpp (hunk 2 above)
Comment (Joseph-Edwards) [reply src/order.cpp:139]: Ah my bad. I'll change it in this PR and open an issue to get rid of it elsewhere (this isn't the first time I've included them).

**Pull request merged**